File: backend/src/main/java/com/crimemanagement/service/UserService.java
package com.crimemanagement.service;

import com.crimemanagement.model.User;
import com.crimemanagement.repository.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import java.util.HashMap;
import java.util.Map;
import java.util.Optional;

@Service
public class UserService {
    
    @Autowired
    private UserRepository userRepository;
    
    public Map<String, Object> authenticateUser(String username, String password) {
        Map<String, Object> response = new HashMap<>();
        
        Optional<User> userOpt = userRepository.findByUsername(username);
        
        if (userOpt.isPresent() && userOpt.get().getPassword().equals(password)) {
            User user = userOpt.get();
            response.put("success", true);
            response.put("user", user);
            response.put("role", user.getRole());
        } else {
            response.put("success", false);
            response.put("message", "Invalid credentials");
        }
        
        return response;
    }
    
    public User registerUser(User user) {
        if (!userRepository.existsByUsername(user.getUsername()) && 
            !userRepository.existsByEmail(user.getEmail())) {
            return userRepository.save(user);
        }
        return null;
    }
    
    public User getUserById(String id) {
        return userRepository.findById(id).orElse(null);
    }
}

File: services/user.service.js
const User = require("../models/user.model")

const userService = {
  authenticateUser: async (username, password) => {
    const response = {}
    const user = await User.findOne({ username })
    console.log("User found in DB:", user ? user.username : "None")
    console.log("Password match:", user && user.password === password)

    if (user && user.password === password) {
      response.success = true
      response.user = user
      response.role = user.role
    } else {
      response.success = false
      response.message = "Invalid credentials"
    }
    return response
  },

  registerUser: async (userData) => {
    const existingUserByUsername = await User.exists({ username: userData.username })
    const existingUserByEmail = await User.exists({ email: userData.email })

    if (!existingUserByUsername && !existingUserByEmail) {
      const newUser = new User(userData)
      return newUser.save()
    }
    return null
  },

  getUserById: async (id) => {
    return User.findById(id)
  },
}

module.exports = userService
